ouroboros-consensus-cardano-test/test/Test/Consensus/Cardano/Translation.hs

```diff
@@ -0,0 +1,226 @@
+import           Ouroboros.Consensus.Cardano.CanHardFork ()
+import           Ouroboros.Consensus.HardFork.Combinator (EraTranslation,
+                     InPairs (..), hardForkEraTranslation, translateLedgerState)
+import           Ouroboros.Consensus.HardFork.Combinator.State.Types
+                     (TranslateLedgerState (translateLedgerStateWith))
+import           Ouroboros.Consensus.HardFork.Combinator.Util.InPairs
+                     (provideBoth)
+import           Ouroboros.Consensus.Ledger.Basics (ApplyMapKind' (..), DiffMK,
+                     EmptyMK, LedgerConfig, LedgerState)
+import           Ouroboros.Consensus.Protocol.Praos.Common
+                     (MaxMajorProtVer (..))
+import           Ouroboros.Consensus.Protocol.Praos.Translate ()
+import           Ouroboros.Consensus.Protocol.TPraos (TPraos)
+import           Ouroboros.Consensus.Shelley.Eras
+import           Ouroboros.Consensus.Shelley.HFEras ()
+import           Ouroboros.Consensus.Shelley.Ledger (ShelleyBlock,
+                     ShelleyLedgerConfig, mkShelleyLedgerConfig,
+                     shelleyLedgerTables, shelleyUTxOTable)
+import           Ouroboros.Consensus.Shelley.Ledger.SupportsProtocol ()
+import           Ouroboros.Consensus.Storage.LedgerDB.HD (UtxoDiff (..),
+                     UtxoEntryDiff (..), UtxoEntryDiffState (..))
+import           Ouroboros.Consensus.TypeFamilyWrappers
+
+import           Test.Cardano.Ledger.Shelley.Utils (unsafeBoundRational)
+import           Test.Consensus.Byron.Generators (genByronLedgerConfig,
+                     genByronLedgerState)
+import           Test.Consensus.Cardano.MockCrypto (MockCryptoCompatByron)
+import           Test.Consensus.Shelley.Generators ()
+import           Test.Consensus.Shelley.MockCrypto
+import           Test.Util.Time (dawnOfTime)
+
+import qualified Cardano.Ledger.Block as Core
+import qualified Cardano.Ledger.Core as Core
+import qualified Cardano.Ledger.Era as Core
+
+type Crypto = MockCryptoCompatByron
+type Era    = ShelleyEra Crypto
+type Proto  = TPraos Crypto
+
+data TestSetup = TestSetup {
+    setupByronLedgerConfig   :: LedgerConfig ByronBlock
+  , setupShelleyLedgerConfig :: LedgerConfig (ShelleyBlock Proto Era)
+  , setupByronLedgerState    :: LedgerState ByronBlock EmptyMK
+}
+
+instance Show TestSetup where
+  show TestSetup {..} = "TestSetup" <> show setupByronLedgerConfig
+                                    <> " "
+                                    <> show setupByronLedgerState
+
+instance Arbitrary TestSetup where
+  arbitrary =
+    TestSetup <$> genByronLedgerConfig
+              <*> pure fixedShelleyLedgerConfig
+              <*> genByronLedgerState
+
+tests :: TestTree
+tests = testGroup "ByronToShelley" [
+    testProperty "test" prop_byronUtxosMatchThoseInShelleyTable
+  ]
+
+prop_byronUtxosMatchThoseInShelleyTable :: TestSetup -> Property
+prop_byronUtxosMatchThoseInShelleyTable TestSetup {..} =
+  let
+    epochNo = EpochNo 10
```

Comment:
What is the effect of the epoch number in this property, and/or in the ledger state translation?